tests/nn/data_parallel/test_fsdp.py

```diff
@@ -305,6 +314,24 @@ def test_cpu_offload_and_cpu_grads(self):
         )
         spawn_and_init(test_fn)
 
+    def test_no_cpu_offload_with_sharded_grad_scaler(self):
+        # We don't test the False condition because that requires the optimizer to internally do
```

Comment:
nit: I know you did not add it but I wonder if we need this comment duplicated in all three tests. Also if could mention what the False property is it would make it a lot clearer.

Reply:
updated comments to reflect what we are testing now.